dnsutil/zone.go

```diff
@@ -0,0 +1,209 @@
+package dnsutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+type zoneGenData struct {
+	Origin     string
+	DefaultTtl uint32
+	Records    []dns.RR
+}
+
+func (z *zoneGenData) Len() int      { return len(z.Records) }
+func (z *zoneGenData) Swap(i, j int) { z.Records[i], z.Records[j] = z.Records[j], z.Records[i] }
+func (z *zoneGenData) Less(i, j int) bool {
+	a, b := z.Records[i], z.Records[j]
```

Comment:
Maybe there should be WriteZoneFile should accept either constant for type of label ordering or comparator. I've encountered needs to have zones written in odd order before.  

Also I don't really know if it's useful but another way of writing zonefiles is by grouping things under $ORIGIN subsections of zone.